app/src/main/java/com/xinto/opencord/ui/viewmodel/MentionsViewModel.kt

```diff
@@ -0,0 +1,102 @@
+package com.xinto.opencord.ui.viewmodel
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.viewModelScope
+import androidx.paging.*
+import com.xinto.opencord.domain.message.DomainMessage
+import com.xinto.opencord.domain.message.toDomain
+import com.xinto.opencord.manager.PersistentDataManager
+import com.xinto.opencord.manager.ToastManager
+import com.xinto.opencord.rest.service.DiscordApiService
+import com.xinto.opencord.ui.viewmodel.base.BasePersistenceViewModel
+import kotlinx.coroutines.flow.emptyFlow
+
+class MentionsViewModel(
+    persistentDataManager: PersistentDataManager,
+    private val toasts: ToastManager,
+    val api: DiscordApiService,
+) : BasePersistenceViewModel(persistentDataManager) {
+    var includeRoles by mutableStateOf(true)
+        private set
+    var includeEveryone by mutableStateOf(true)
+        private set
+    var includeAllServers by mutableStateOf(true)
+        private set
+
+    var messages by mutableStateOf(emptyFlow<PagingData<DomainMessage>>())
```

Comment:
its so you can recreate the flow with different params and it rerenders
i took idea from hyperion

Reply:
This looks awful. It's basically writing `State<State<DomainMessage>>`. Also don't take such ideas from Hyperion, zt doesn't understand coroutines and flows

Reply:
i dont see another way

Reply:
What exactly are you trying to achieve?

Reply:
Make the UI update when I recreate the pager in `initPager()` (toggle role/@everyone/all servers)

Reply:
You could probably use AndroidX Paging's refresh key instead. Although I'm not 100% sure because paging3 fucking sucks. If there's no way then I guess we could leave it as it is now

Reply:
> This looks awful. It's basically writing `State<State<DomainMessage>>`. Also don't take such ideas from Hyperion, zt doesn't understand coroutines and flows

also i dont see how its double state
flow is not a state otherwise reassigning it would've worked
Its `MutableState<Flow<PagingData<DomainMessage>>>`

Reply:
it's double state because Flow is basically State but for Kotlin coroutines, plus compose has an extension on Flows to convert them to states: `Flow<T>.collectAsState()`

Reply:
Again, theres no way to notify compose that the Flow itself changed (not it's content) without surrounding the flow in a state

Reply:
I mean ideally, you shouldn't need to change the flow itself but rather just change its content, which should be done with the refresh key of AndroidX paging, but if that doesn't work then we *can* allow this, at least temporarily

Reply:
If I want to change the paging source (and its params) then I can:
- recreate it and the pager that owns it
- make it an inner class of MentionsViewModel, making it tied to the viewmodel and use the params directly from the state variables and only then refresh the pager
IMO this is way more horrendous than just recreating the pager(flow)

Reply:
anyways, i guess this is fine for now, no?

Reply:
I guess